packages/charts/react-charting/src/components/ScatterChart/ScatterChart.base.tsx

```diff
@@ -392,21 +394,81 @@ export const ScatterChartBase: React.FunctionComponent<IScatterChartProps> = Rea
     [_points, props.culture, props.useUTC],
   );
 
-  function _getRangeForScatterMarkerSize(
-    yScale: ScaleLinear<number, number>,
-    yPadding: number,
-    xMin: number,
-    xMax: number,
-    xPadding: number,
-  ): number {
-    const extraXPixels = getRTL()
-      ? _xAxisScale.current?.(xMax - xPadding) - _xAxisScale.current?.(xMax)
-      : _xAxisScale.current?.(xMin + xPadding) - _xAxisScale.current?.(xMin);
-
-    const yMin = yScale.domain()[0];
-    const extraYPixels = yScale(yMin) - yScale(yMin + yPadding);
-    return Math.min(extraXPixels, extraYPixels);
-  }
+  const _getRangeForScatterMarkerSize = React.useCallback(
+    (yScale: ScaleLinear<number, number>, yPadding: number, xMin: number, xMax: number, xPadding: number): number => {
+      const extraXPixels = getRTL()
+        ? _xAxisScale.current?.(xMax - xPadding) - _xAxisScale.current?.(xMax)
+        : _xAxisScale.current?.(xMin + xPadding) - _xAxisScale.current?.(xMin);
+
+      const yMin = yScale.domain()[0];
+      const extraYPixels = yScale(yMin) - yScale(yMin + yPadding);
+      return Math.min(extraXPixels, extraYPixels);
+    },
+    [],
+  );
+
+  const _renderShape = React.useCallback(
+    (
+      shape: LegendShape | undefined,
+      x: number | string | Date,
+      y: number,
+      size: number,
+      fill: string,
+      stroke: string,
+      opacity: number,
+      circleId: string,
+      isLegendSelected: boolean,
+      eventHandlers: any,
+      ariaLabel: string,
+      tabIndex?: number,
+    ): JSX.Element => {
+      const cx = _xAxisScale.current?.(x) + _xBandwidth.current;
+      const cy = _yAxisScale.current?.(y);
+
+      // Default to circle if no shape specified
+      if (!shape || shape === 'default' || shape === 'circle') {
+        return (
+          <circle
```

Comment:
This circle can also be defined using path. It will eliminate the need to define a separate function and prevent code duplication in argument passing and common attributes. This can be taken as a follow-up.